### PR TITLE
Fix: User menu bar pushed down by calamity banner 

### LINKED ISSUE #2786: User menu bar pushed down by calamity banner
## Steps to reproduce

1. Ensure that the calamity banner is visible in the app.
2. Open up the action panel.
3. Note the position of the user menu bar in the header.

## Expected behaviour

* The action panel should not sit over the calamity banner and should move down with the rest of the page.

## Actual behaviour

* Currently the user menu bar is misaligned in the calamity banner due to the presence of the calamity banner.

image

image

user-menu-bar

image

image

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

See #2786 for original issue description.

In addition to fixing the bug above, this PR also fixes another bug with `Pending` transaction button not triggering the user hub to open and refactors the way user hub open state is managed.

https://github.com/user-attachments/assets/74b259bd-49a5-4232-8a23-98a3ccfd97a5

## Testing

To confirm the fix works, you'll need to get the calamity banner to show. The easiest way is to modify line `128` of `src/components/frame/Extensions/layouts/ColonyLayout.tsx` and change `canUpgrade` to `true` so that it always displays the banner.

Please also test other user hub interactions. This includes:
- Opening User Hub from different places (user nav, action sidebar)
- Interacting with User Hub when tokens modal is open
- Interacting with User Hub on mobile/tablet
- Navigating between tabs
- Clicking on TxButton opening the Transactions tab of User Hub

Resolves #2786

### CODE REVIEW

File: src/components/frame/Extensions/layouts/ColonyLayout.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ const ColonyLayout: FC<PropsWithChildren> = ({ children }) => {
         txButton={<TxButton />}
         userHub={<UserHubButton />}
         className={clsx(
-          'modal-blur-navigation [.show-header-in-modal_&]:z-userNavModal',
+          'modal-blur-navigation [.show-header-in-modal_&]:absolute [.show-header-in-modal_&]:right-6 [.show-header-in-modal_&]:top-8 [.show-header-in-modal_&]:z-userNavModal',
```
Comment (jakubcolony) [inline src/components/frame/Extensions/layouts/ColonyLayout.tsx:97]: src/components/frame/Extensions/layouts/ColonyLayout.tsx

File: src/components/common/Extensions/UserHubButton/UserHubButton.tsx [hunk 2]
```diff
@@ -84,40 +76,46 @@ const UserHubButton: FC = () => {
           {
             name: 'offset',
             options: {
-              offset: popperTooltipOffset,
+              offset: isMobile ? [0, 1] : [0, 8],
```
Comment (rdig) [inline src/components/common/Extensions/UserHubButton/UserHubButton.tsx:79]: Ufff.

I know this was the "way" to make this work, but god damn these might as well be "magic" numbers at this point :)

**Review by rdig — APPROVED** [review]
All good as far as I could test. I tried user hub from different places, action sidebar open, desktop and mobile.

Screenshot from 2024-07-30 00-10-59
Screenshot from 2024-07-30 00-11-53
Screenshot from 2024-07-30 00-12-01
Screenshot from 2024-07-30 00-12-16
Screenshot from 2024-07-30 00-13-02
Screenshot from 2024-07-30 00-13-11
Screenshot from 2024-07-30 00-13-15
Screenshot from 2024-07-30 00-13-34

The only interaction I found "questionable", was when I was in mobile view, having a action open, then opening the user hub. There's like 20px of space between the top header and the user hub, from which the action view below renders. 

It's a bit hard to explain, so here's a view showcasing this:

[Screencast from 2024-07-30 00-14-19.webm](https://github.com/user-attachments/assets/450fc80d-3a7c-413f-bf7e-983dc65c68ee)

I don't actually know if this is a design decision or not, but personally I'd catalog this as a bug.

File: src/components/frame/Extensions/layouts/ColonyLayout.tsx (hunk 1 above)
Comment (jakubcolony) [inline src/components/frame/Extensions/layouts/ColonyLayout.tsx:97]: This is not great in terms of readability, if you have any suggestions how to make it better please share!

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this so far, the refactor all looks good! The userhub is now in the correct place in the action sidebar.

Tested the pending button opening the userhub which is working nicely:

https://github.com/user-attachments/assets/444f1175-60d3-4295-b69d-89864108f310

There is an issue with the tokens modals though causing the app to shift around. This is happening on desktop and tablet.

https://github.com/user-attachments/assets/1200b807-fa9d-4f1c-b8c8-367317683122

 I've observed the same bug as Raul when opening the userhub with the action sidebar open on mobile.

https://github.com/user-attachments/assets/ade7db3e-7f01-47b5-8e92-1014cdfaf889

I'm not sure if this is an issue on master, or has been introduced here, but the userhub doesn't automatically open on the transactions hub after a failed transaction:

https://github.com/user-attachments/assets/33001372-f0ea-43eb-8ddc-84b6cd073118

**Comment by melyndav** [comment]
> I don't actually know if this is a design decision or not, but personally I'd catalog this as a bug.

This is a bug. The design system component for the user hub shows it at full screen - https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1782-50431&t=Yvc02UhER4ZOocon-4

**Comment by jakubcolony** [comment]
Closing this PR as the original issue will be resolved in navigation UI refactor by @rumzledz.

**Pull request closed without merging**